quantum/pointing_device/pointing_device_modes.c

```diff
@@ -0,0 +1,658 @@
+        pointing_mode_context.config.tg_mode_id = POINTING_MODE_DEFAULT;
+    } else {
+        pointing_mode_context.config.tg_mode_id = mode_id;
+    }
+    if (pointing_mode_context.mode.id != pointing_mode_context.config.tg_mode_id) pointing_mode_reset();
+}
+
+/**
+ * @brief Access current toggled pointing mode
+ *
+ * @return uint8_t toggle pointing mode
+ */
+uint8_t get_toggled_pointing_mode_id(void) {
+    return pointing_mode_context.config.tg_mode_id;
+}
+
+/**
+ * @brief Weak function to convert x/y axes to h/v
+ *
+ * The default uses accumulation based on inversion defines and
+ * halts cursor movement
+ *
+ * @params pointing_mode[in] pointing_mode_t
+ * @params mouse_report[in]  report_mouse_t
+ *
+ * @return updated mouse_report report_mouse_t
+ */
+__attribute__((weak)) report_mouse_t pointing_modes_axes_conv(pointing_mode_t pointing_mode, report_mouse_t mouse_report) {
+#    ifdef POINTING_DEVICE_MODES_INVERT_X
+    pointing_mode.x -= mouse_report.x;
+#    else
+    pointing_mode.x += mouse_report.x;
+#    endif
+#    ifdef POINTING_DEVICE_MODES_INVERT_Y
+    pointing_mode.y -= mouse_report.y;
+#    else
+    pointing_mode.y += mouse_report.y;
+#    endif
+    set_pointing_mode(pointing_mode);
+    mouse_report.x = 0;
+    mouse_report.y = 0;
+    return mouse_report;
+}
+
+/**
+ * @brief Modifies divisor after
+ *
+ * @params pointing_mode[in] uint8_t
+ * @params direction[in] uint8_t
+ *
+ * @return divisor uint8_t
+ */
+static uint8_t divisor_postprocess(uint8_t divisor) {
+    if (!(pointing_mode_divisor_postprocess_user(&divisor) && pointing_mode_divisor_postprocess_kb(&divisor))) {
+        return divisor;
+    }
+    // Modify divisor if precision is toggled
+    if (get_toggled_pointing_mode_id() == PM_PRECISION && !(get_pointing_mode_id() == PM_PRECISION)) {
+        divisor = ((uint16_t)divisor * POINTING_PRECISION_DIVISOR) > UINT8_MAX ? UINT8_MAX : (divisor * POINTING_PRECISION_DIVISOR);
+    }
+    // Prevent 0 divisor
+    if (!divisor) divisor = POINTING_DEFAULT_DIVISOR;
+    return divisor;
+}
+
+/**
+ * @brief local function to get pointing mode divisor
+ *
+ * Will handle default divisors and call weak kb and user functions
+ *
+ * NOTE: Expects that pointing mode and direction has been updated
+ *
+ * @return divisor uint8_t
+ */
+static uint8_t get_pointing_mode_divisor(void) {
+    uint8_t divisor = 0;
+    // allow for user and keyboard overrides
+    divisor = get_pointing_mode_divisor_user(get_pointing_mode_id(), pointing_mode_context.mode.direction);
+    if (divisor) return divisor_postprocess(divisor);
+    divisor = get_pointing_mode_divisor_kb(get_pointing_mode_id(), pointing_mode_context.mode.direction);
+    if (divisor) return divisor_postprocess(divisor);
+    // built in divisors
+    switch (get_pointing_mode_id()) {
+        case PM_PRECISION:
+            divisor = POINTING_PRECISION_DIVISOR;
+            break;
+
+        case PM_DRAG:
+            divisor = POINTING_DRAG_DIVISOR_H;
```

Comment:
```suggestion
            divisor = POINTING_DRAG_DIVISOR;
```

Reply:
Thanks for catching that no idea how that got by my compile testing must have had `POINTING_DRAG_DIVISOR_H` defined somewhere